plugin/discovery/get.go

```diff
@@ -0,0 +1,170 @@
+package discovery
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"runtime"
+	"strings"
+
+	"golang.org/x/net/html"
+
+	getter "github.com/hashicorp/go-getter"
+)
+
+const releasesURL = "https://releases.hashicorp.com/"
+
+// pluginURL generates URLs to lookup the versions of a plugin, or the file path.
```

Comment:
sorry, that was a mixed up rebase. This get renamed correctly in a later commit.